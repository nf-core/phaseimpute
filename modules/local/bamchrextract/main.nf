process BAMCHREXTRACT {
    tag "$meta.id"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/samtools:1.20--h50ea8bc_0' :
        'biocontainers/samtools:1.20--h50ea8bc_0' }"

    input:
    tuple val(meta), path(input)

    output:
    tuple val(meta), path("*.txt"), emit: chr
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    samtools \\
        head \\
        $input \| \\
        grep '^@SQ' | cut -d$'\t' -f2 \\
        > ${prefix}.txt


    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        samtools: \$( samtools --version |& sed '1!d; s/^.*samtools //' )
        grep: \$( grep --version |& grep -o -E '[0-9]+\\.[0-9]+' )
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        samtools: \$( samtools --version |& sed '1!d; s/^.*samtools //' )
        grep: \$( grep --help |& grep -o -E '[0-9]+\\.[0-9]+\\.[0-9]+' )
    END_VERSIONS
    """
}
