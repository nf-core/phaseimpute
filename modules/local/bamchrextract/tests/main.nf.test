nextflow_process {

    name "Test Process BAMCHREXTRACT"
    script "../main.nf"
    process "BAMCHREXTRACT"

    tag "modules"
    tag "modules_local"
    tag "bamchrextract"

    test("Extract chr from bam") {
        when {
            config "./nextflow.config"
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    file(params.test_data['sarscov2']['illumina']['test_single_end_bam'], checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
