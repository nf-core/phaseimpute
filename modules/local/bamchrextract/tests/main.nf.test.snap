{
    "Extract chr from bam": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "chr": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-06-16T12:06:26.625434359"
    }
}